### PR TITLE
Add Kotlin Multiplatform project output

### PR DESCRIPTION
**Pull request by sleeyax**
Changes:
- Convert regular Android project to Kotlin Multiplatform output (`common` and `android` sourcesets) so it works in both KMM as well as a regular Android project
- Setup moko resources to target both Android and iOS (future) targets. 
- Adjust `jitpack.yml` accordingly

### CODE REVIEW

File: jitpack.yml [hunk 1]
```diff
@@ -1,19 +1,13 @@
 env:
-  NODE_VERSION: "12.22.1"
+  NODE_VERSION: "14.15.0"
   NODE_SHA256: "d315c5dea4d96658164cdb257bd8dbb5e44bdd2a7c1d747841f06515f23a0042"
-  GRADLE_VERSION: "6.5"
   GRADLE_SHA256: "23e7d37e9bb4f8dabb8a3ea7fdee9dd0428b9b1a71d298aefd65b11dccea220f"
```
Comment (TheBeastLT) [inline jitpack.yml:4]: Can also remove this one, as it's not needed anymore

**Comment by sleeyax** [comment]
One more thing: as Tim suggested we should be able to merge KMM and Android build scripts into one. Moko generates vector XML files to `androidMain/res/drawable` from its input SVGs, meaning we ultimately get duplicate outputs. We should verify that the generated vecotors match the vectors outputed by the current android build script though.

That being said, even without the above improvement this PR should be ready for use in both KMM as well as regular Android projects.

File: android/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+group = "com.github.Stremio"
+version = "5.0.0"
+
+buildscript {
+    repositories {
+        gradlePluginPortal()
+    }
+
+    dependencies {
+        classpath("dev.icerock.moko:resources-generator:0.23.0")
```
Comment (TheBeastLT) [inline android/build.gradle.kts:10]: Can you reuse the version here aswell?

File: jitpack.yml [hunk 3]
```diff
@@ -1,19 +1,14 @@
 env:
-  NODE_VERSION: "12.22.1"
+  NODE_VERSION: "14.15.0"
   NODE_SHA256: "d315c5dea4d96658164cdb257bd8dbb5e44bdd2a7c1d747841f06515f23a0042"
-  GRADLE_VERSION: "6.5"
-  GRADLE_SHA256: "23e7d37e9bb4f8dabb8a3ea7fdee9dd0428b9b1a71d298aefd65b11dccea220f"
+jdk:
+  - openjdk11
```
Comment (TheBeastLT) [inline jitpack.yml:5]: Why is this needed? Will we be able to run this then on lower android versions if this lib will be compiled against java 11?

Comment (sleeyax) [reply jitpack.yml:5]: Because without this there's the following build error ([full log](https://jitpack.io/com/github/Stremio/stremio-icons/92ed5fee2c/build.log)):

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'build'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve dev.icerock.moko:resources-generator:0.23.0.
     Required by:
         project :
         project : > dev.icerock.mobile.multiplatform-resources:dev.icerock.mobile.multiplatform-resources.gradle.plugin:0.23.0
      > No matching variant of dev.icerock.moko:resources-generator:0.23.0 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.4' but:
          - Variant 'apiElements' capability dev.icerock.moko:resources-generator:0.23.0 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4')
          - Variant 'javadocElements' capability dev.icerock.moko:resources-generator:0.23.0 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4')
          - Variant 'runtimeElements' capability dev.icerock.moko:resources-generator:0.23.0 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4')
          - Variant 'sourcesElements' capability dev.icerock.moko:resources-generator:0.23.0 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4')
```

> Will we be able to run this then on lower android versions if this lib will be compiled against java 11?

Not sure of possible complications tbh. I assumed this was just a build time/packaging thing.

Feel free to suggest an alternative solution.

File: android/build.gradle.kts (hunk 2 above)
Comment (sleeyax) [reply android/build.gradle.kts]: Jep, good catch.

File: jitpack.yml (hunk 3 above)
Comment (TheBeastLT) [reply jitpack.yml:5]: If it's a requirement then not sure if an alternative exists. Maybe try going with this, but then make sure that the app works fine on min sdk version of android.

**Review by TheBeastLT — APPROVED** [review]

**Review by tymmesyde — APPROVED** [review]

**Comment by sleeyax** [comment]
The [latest build](https://jitpack.io/com/github/Stremio/stremio-icons/8fe237f367/build.log) is succeeding so LGTM :rocket:
